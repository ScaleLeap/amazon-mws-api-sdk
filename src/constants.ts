/**
 * The version value is automatically replaced during builds.
 */
export const USER_AGENT = '@scaleleap/amazon-mws-api-sdk/1.9.0'
