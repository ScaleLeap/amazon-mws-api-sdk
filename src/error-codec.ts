import { Codec, exactly, oneOf, optional, string } from 'purify-ts/Codec'

export enum CodeEnum {
  InputStreamDisconnected,
  InvalidParameterValue,
  AccessDenied,
  InvalidAccessKeyId,
  SignatureDoesNotMatch,
  InvalidAddress,
  InternalError,
  'Internal Error',
  QuotaExceeded,
  RequestThrottled,
  ResourceNotFound,
  ScheduledPackageAlreadyExists,
  RegionNotSupported,
  ScheduleWindowExpired,
  InvalidOrderState,
  PickupSlotNotAvailable,
  AccessToFeedProcessingResultDenied,
  ContentMD5Missing,
  ContentMD5DoesNotMatch,
  FeedCanceled,
  FeedProcessingResultNoLongerAvailable,
  FeedProcessingResultNotReady,
  InputDataError,
  InvalidFeedSubmissionId,
  InvalidFeedType,
  InvalidRequest,
  NonRetriableInternalError,
  RetriableInternalError,
  AccessToReportDenied,
  InvalidReportId,
  InvalidReportRequestId,
  InvalidReportType,
  InvalidScheduleFrequency,
  ReportNoLongerAvailable,
  ReportNotReady,
  DependencyFatalException,
  DependencyRetriableException,
  DependencyUnauthorizedException,
  InternalErrorFatalException,
  InvalidInputFatalException,
}

export const Error = Codec.interface({
  Type: string,
  Code: oneOf(Object.keys(CodeEnum).map((x) => exactly(x)) as [Codec<string>, ...Codec<string>[]]),
  Message: string,
  Detail: optional(string),
})

export const MWSApiError = Codec.interface({
  ErrorResponse: Codec.interface({
    Error,
    RequestID: optional(string),
    RequestId: optional(string),
  }),
})
