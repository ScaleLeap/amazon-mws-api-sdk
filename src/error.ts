/* eslint-disable max-classes-per-file */
export class MWSError extends Error {
  constructor(...parameters: string[]) {
    // Propagate some vendor-specific arguments
    super(...parameters)

    // If someone downlevels the compilation target to es5
    Object.setPrototypeOf(this, MWSError.prototype)

    // Maintains proper stack trace (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, MWSError)
    }
  }
}

export class HttpError extends MWSError {
  public message = 'Encountered an error while sending a request'

  constructor(public error: unknown, ...parameters: string[]) {
    super(...parameters)
    Object.setPrototypeOf(this, HttpError.prototype)

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, HttpError)
    }
  }
}

export class ParsingError extends MWSError {
  public message = 'Encountered an error while parsing a response'

  constructor(public error: string, ...parameters: string[]) {
    super(...parameters)
    Object.setPrototypeOf(this, ParsingError.prototype)

    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, ParsingError)
    }
  }
}
/* eslint-enable max-classes-per-file */
