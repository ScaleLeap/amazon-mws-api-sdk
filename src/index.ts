export * from './http'
export * from './mws'
export * from './sections/sellers'
export * from './sections/orders'
export * from './sections/products/products'
export * from './sections/fulfillment-inventory'
export * from './sections/reports'
export * from './error'
export * from '@scaleleap/amazon-marketplaces'
