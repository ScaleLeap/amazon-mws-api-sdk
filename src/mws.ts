import { HttpClient } from './http'
import { FulfillmentInventory } from './sections/fulfillment-inventory'
import { Orders } from './sections/orders'
import { Products } from './sections/products/products'
import { Reports } from './sections/reports'
import { Sellers } from './sections/sellers'

export class MWS {
  private _sellers!: Sellers

  private _orders!: Orders

  private _fulfillmentInventory!: FulfillmentInventory

  private _products!: Products

  private _reports!: Reports

  constructor(private httpClient: HttpClient) {}

  get sellers() {
    if (!this._sellers) {
      this._sellers = new Sellers(this.httpClient)
    }

    return this._sellers
  }

  get orders() {
    if (!this._orders) {
      this._orders = new Orders(this.httpClient)
    }

    return this._orders
  }

  get fulfillmentInventory() {
    if (!this._fulfillmentInventory) {
      this._fulfillmentInventory = new FulfillmentInventory(this.httpClient)
    }

    return this._fulfillmentInventory
  }

  get products() {
    if (!this._products) {
      this._products = new Products(this.httpClient)
    }

    return this._products
  }

  get reports() {
    if (!this._reports) {
      this._reports = new Reports(this.httpClient)
    }

    return this._reports
  }
}
