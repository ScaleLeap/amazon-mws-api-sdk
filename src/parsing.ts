/** A collection of parsing utils */

export const ensureArray = <T>(x: T | T[]): T[] => (Array.isArray(x) ? x : [x])

export const parseBoolean = (x: 'Yes' | 'No'): boolean => {
  switch (x) {
    case 'Yes':
      return true
    case 'No':
      return false
    default:
      throw new Error('TODO, but should be library-specific error')
  }
}
