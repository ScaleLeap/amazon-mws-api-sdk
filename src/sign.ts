import crypto from 'crypto'

export const sign = (queryString: string, secret: string): string => {
  const hmac = crypto.createHmac('sha256', secret)
  hmac.update(queryString, 'utf8')
  return hmac.digest('base64')
}
