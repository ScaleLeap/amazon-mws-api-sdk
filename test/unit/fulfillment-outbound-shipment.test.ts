import { canonicalizeUpdateFulfillmentOrderParameters, NextToken, ParsingError } from '../../src'
import {
  canonicalizeGetFulfillmentPreviewParameters,
  CreateFulfillmentOrderParameters,
} from '../../src/sections/fulfillment-outbound-shipment/type'
import { createMockHttpClient, mockMwsFail, mockMwsServiceStatus, parsingError } from '../utils'

function mockEnum() {
  /**
   * Mock everything in purify-ts, restore it back to normal,
   *  except for `enumeration` which will be stubbed to accept
   * https://github.com/facebook/jest/issues/936#issuecomment-265074320
   */
  const original = jest.requireActual('purify-ts')

  return {
    ...original, // Pass down all the exported objects
    enumeration: <T extends Record<string, string | number>>(enumeration: T) => {
      const enumValues = Object.values(enumeration)

      return original.Codec.custom({
        decode: (input: string | number) => {
          if (typeof input !== 'string' && typeof input !== 'number') {
            return original.Left(`Expected enum, received ${input}`)
          }

          const enumIndex = enumValues.indexOf(input)

          return enumIndex !== -1 || input === 'String'
            ? original.Right((enumValues[enumIndex] as T[keyof T]) || 'String')
            : original.Left(`Expected enum, received ${input}`)
        },
        encode: original.identity,
        schema: () => ({ enum: enumValues }),
      })
    },
  }
}
jest.mock('purify-ts', () => mockEnum())

const mockAddress = {
  Name: '',
  Line1: '',
  Line2: '',
  Line3: '',
  DistrictOrCounty: '',
  City: '',
  StateOrProvinceCode: '',
  CountryCode: '',
  PostalCode: '',
  PhoneNumber: '',
}

const mockCreateFulfillmentOrderItem = {
  SellerSKU: '',
  SellerFulfillmentOrderItemId: '',
  Quantity: 1,
}

const mockGetFulfillmentPreviewItem = {
  SellerSKU: '',
  SellerFulfillmentOrderItemId: '',
  Quantity: 1,
}

describe('fulfillmentOutboundShipment', () => {
  describe('parameters', () => {
    describe('canonicalizeGetFulfillmentPreviewParameters', () => {
      it('should properly canonicalize GetFulfillmentPreviewParameters', () => {
        expect.assertions(1)

        const parameters = {
          Address: mockAddress,
          Items: [mockGetFulfillmentPreviewItem],
        }

        const output = canonicalizeGetFulfillmentPreviewParameters(parameters)

        expect(output['Items.member']).toStrictEqual(parameters.Items)
      })
    })

    describe('canonicalizeUpdateFulfillmentOrderParameters', () => {
      it('should properly canonicalize UpdateFulfillmentOrderParameters', () => {
        expect.assertions(2)

        const mockItem = {
          SellerFulfillmentOrderItemId: '',
          Quantity: 1,
        }

        const parameters = {
          SellerFulfillmentOrderId: '',
          NotificationEmailList: [''],
          Items: [mockItem],
        }

        const output = canonicalizeUpdateFulfillmentOrderParameters(parameters)

        expect(output['Items.member']).toStrictEqual(parameters.Items)
        expect(output['NotificationEmailList.member']).toStrictEqual(
          parameters.NotificationEmailList,
        )
      })
    })
  })

  describe('createFulfillmentReturn', () => {
    const mockCreateReturnItem = {
      SellerReturnItemId: '',
      SellerFulfillmentOrderItemId: '',
      AmazonShipmentId: '',
      ReturnReasonCode: '',
    }

    const parameters = {
      SellerFulfillmentOrderId: '',
      Items: [mockCreateReturnItem],
    }

    it('returns the list of invalid and valid items for return if succesful', async () => {
      expect.assertions(1)

      const mockCreateFulfillmentReturn = createMockHttpClient(
        'fulfillment_outbound_shipment_create_fulfillment_return',
      )

      expect(
        await mockCreateFulfillmentReturn.fulfillmentOutboundShipment.createFulfillmentReturn(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response isnt  valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.createFulfillmentReturn(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('listReturnReasonCodes', () => {
    const parameters = {
      SellerFulfillmentOrderId: '',
      SellerSKU: '',
    }

    it('returns a list of reason code details if succesful', async () => {
      expect.assertions(1)

      const mockListReturnReasonCodes = createMockHttpClient(
        'fulfillment_outbound_shipment_list_return_reason_codes',
      )

      expect(
        await mockListReturnReasonCodes.fulfillmentOutboundShipment.listReturnReasonCodes(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response  isnt valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.listReturnReasonCodes(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('cancelFulfillmentOrder', () => {
    const parameters = { SellerFulfillmentOrderId: '' }

    it('returns the standard response if succesful', async () => {
      expect.assertions(1)

      const mockCancelFulfillmentOrder = createMockHttpClient(
        'fulfillment_outbound_shipment_cancel_fulfillment_order',
      )

      expect(
        await mockCancelFulfillmentOrder.fulfillmentOutboundShipment.cancelFulfillmentOrder(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response  isnt valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.cancelFulfillmentOrder(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('getPackageTrackingDetails', () => {
    const parameters = { PackageNumber: 0 }

    it('returns package tracking details if succesful', async () => {
      expect.assertions(1)

      const mockGetPackageTrackingDetails = createMockHttpClient(
        'fulfillment_outbound_shipment_get_package_tracking_details',
      )

      expect(
        await mockGetPackageTrackingDetails.fulfillmentOutboundShipment.getPackageTrackingDetails(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response i snt valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.getPackageTrackingDetails(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('listAllFulfillmentOrdersByNextToken', () => {
    const mockNextToken = new NextToken('ListAllFulfillmentOrders', '123')

    it('returns a list of fulfillment orders if succesful', async () => {
      expect.assertions(1)

      const mockListAllFulfillmentOrders = createMockHttpClient(
        'fulfillment_outbound_shipment_list_all_fulfillment_orders_nt',
      )

      expect(
        await mockListAllFulfillmentOrders.fulfillmentOutboundShipment.listAllFulfillmentOrdersByNextToken(
          mockNextToken,
        ),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response i snt valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.listAllFulfillmentOrdersByNextToken(mockNextToken),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('getFulfillmentOrder', () => {
    const parameters = { SellerFulfillmentOrderId: '' }

    it('succesfully parses response structure, from c sharp', async () => {
      expect.assertions(1)

      const mockGetFulfillmentOrder = createMockHttpClient(
        'fulfillment_outbound_shipment_get_fulfillment_order_from_c_sharp',
      )

      expect(
        await mockGetFulfillmentOrder.fulfillmentOutboundShipment.getFulfillmentOrder(parameters),
      ).toMatchSnapshot()
    })

    it('returns a fulfillment order based on a SellerFulfillmentOrderId if succesful', async () => {
      expect.assertions(1)

      const mockGetFulfillmentOrder = createMockHttpClient(
        'fulfillment_outbound_shipment_get_fulfillment_order',
      )

      expect(
        await mockGetFulfillmentOrder.fulfillmentOutboundShipment.getFulfillmentOrder(parameters),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response is nt valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.getFulfillmentOrder(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('listAllFulfillmentOrders', () => {
    it('returns a list of fulfillment orders if succesful', async () => {
      expect.assertions(1)

      const mockListAllFulfillmentOrders = createMockHttpClient(
        'fulfillment_outbound_shipment_list_all_fulfillment_orders',
      )

      expect(
        await mockListAllFulfillmentOrders.fulfillmentOutboundShipment.listAllFulfillmentOrders(),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response is nt valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.listAllFulfillmentOrders(),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('updateFulfillmentOrder', () => {
    const parameters = {
      SellerFulfillmentOrderId: '',
    }

    it('returns the standard response elements if succesful', async () => {
      expect.assertions(1)

      const mockUpdateFulfillmentOrder = createMockHttpClient(
        'fulfillment_outbound_shipment_update_fulfillment_order',
      )

      expect(
        await mockUpdateFulfillmentOrder.fulfillmentOutboundShipment.updateFulfillmentOrder(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response isnt valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.updateFulfillmentOrder(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('createFulfillmentOrder', () => {
    const parameters: CreateFulfillmentOrderParameters = {
      SellerFulfillmentOrderId: '',
      DisplayableOrderId: '',
      DisplayableOrderDateTime: new Date(),
      DisplayableOrderComment: '',
      ShippingSpeedCategory: 'Priority',
      DestinationAddress: mockAddress,
      Items: [mockCreateFulfillmentOrderItem],
    }

    it('returns the standard response elements if succesful', async () => {
      expect.assertions(1)

      const mockCreateFulfillmentOrder = createMockHttpClient(
        'fulfillment_outbound_shipment_create_fulfillment_order',
      )

      expect(
        await mockCreateFulfillmentOrder.fulfillmentOutboundShipment.createFulfillmentOrder(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response isnt valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.createFulfillmentOrder(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('getFulfillmentPreview', () => {
    const parameters = {
      Address: mockAddress,
      Items: [mockGetFulfillmentPreviewItem],
    }

    it('returns a list of fulfillment previews if succesful', async () => {
      expect.assertions(1)

      const mockGetFulfillmentPreview = createMockHttpClient(
        'fulfillment_outbound_shipment_get_fulfillment_preview',
      )

      expect(
        await mockGetFulfillmentPreview.fulfillmentOutboundShipment.getFulfillmentPreview(
          parameters,
        ),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response is not valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.getFulfillmentPreview(parameters),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })

  describe('getServiceStatus', () => {
    it('returns a parsed model when the status response is valid', async () => {
      expect.assertions(1)

      expect(
        await mockMwsServiceStatus.fulfillmentOutboundShipment.getServiceStatus(),
      ).toMatchSnapshot()
    })

    it('throws a parsing error when the status response is not valid', async () => {
      expect.assertions(1)

      await expect(() =>
        mockMwsFail.fulfillmentOutboundShipment.getServiceStatus(),
      ).rejects.toStrictEqual(new ParsingError(parsingError))
    })
  })
})
