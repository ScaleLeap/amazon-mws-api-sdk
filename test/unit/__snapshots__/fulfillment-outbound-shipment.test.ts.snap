// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fulfillmentOutboundShipment getFulfillmentPreview returns a list of fulfillment previews if succesful 1`] = `
Array [
  Object {
    "FulfillmentPreviews": Array [
      Object {
        "EstimatedFees": Array [
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "225",
            },
            "Name": "FBAPerUnitFulfillmentFee",
          },
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "475",
            },
            "Name": "FBAPerOrderFulfillmentFee",
          },
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "660",
            },
            "Name": "FBATransportationFee",
          },
        ],
        "EstimatedShippingWeight": Object {
          "Unit": "kilograms",
          "Value": "12",
        },
        "FulfillmentPreviewShipments": Array [
          Object {
            "EarliestArrivalDate": 2014-01-05T07:00:00.000Z,
            "EarliestShipDate": 2014-01-04T00:30:00.000Z,
            "FulfillmentPreviewItems": Array [
              Object {
                "EstimatedShippingWeight": Object {
                  "Unit": "kilograms",
                  "Value": "10",
                },
                "Quantity": 2,
                "SellerFulfillmentOrderItemId": "mws-test-query-20100713023406723-2",
                "SellerSKU": "SampleSKU1",
                "ShippingWeightCalculationMethod": "Package",
              },
              Object {
                "EstimatedShippingWeight": Object {
                  "Unit": "kilograms",
                  "Value": "2",
                },
                "Quantity": 1,
                "SellerFulfillmentOrderItemId": "mws-test-query-20100713023406723-1",
                "SellerSKU": "SampleSKU2",
                "ShippingWeightCalculationMethod": "Package",
              },
            ],
            "LatestArrivalDate": 2014-01-06T06:59:59.000Z,
            "LatestShipDate": 2014-01-04T00:30:00.000Z,
          },
        ],
        "IsCODCapable": false,
        "IsFulfillable": true,
        "MarketplaceId": "ATVPDKIKX0DER",
        "ShippingSpeedCategory": "Expedited",
        "UnfulfillablePreviewItems": Array [],
      },
      Object {
        "EstimatedFees": Array [
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "225",
            },
            "Name": "FBAPerUnitFulfillmentFee",
          },
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "475",
            },
            "Name": "FBAPerOrderFulfillmentFee",
          },
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "660",
            },
            "Name": "FBATransportationFee",
          },
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "315",
            },
            "Name": "FBAFulfillmentCODFee",
          },
        ],
        "EstimatedShippingWeight": Object {
          "Unit": "kilograms",
          "Value": "12",
        },
        "FulfillmentPreviewShipments": Array [
          Object {
            "EarliestArrivalDate": 2014-01-05T07:00:00.000Z,
            "EarliestShipDate": 2014-01-04T00:30:00.000Z,
            "FulfillmentPreviewItems": Array [
              Object {
                "EstimatedShippingWeight": Object {
                  "Unit": "kilograms",
                  "Value": "10",
                },
                "Quantity": 2,
                "SellerFulfillmentOrderItemId": "mws-test-query-20100713023406723-2",
                "SellerSKU": "SampleSKU1",
                "ShippingWeightCalculationMethod": "Package",
              },
              Object {
                "EstimatedShippingWeight": Object {
                  "Unit": "kilograms",
                  "Value": "2",
                },
                "Quantity": 1,
                "SellerFulfillmentOrderItemId": "mws-test-query-20100713023406723-1",
                "SellerSKU": "SampleSKU2",
                "ShippingWeightCalculationMethod": "Package",
              },
            ],
            "LatestArrivalDate": 2014-07-09T06:59:59.000Z,
            "LatestShipDate": 2014-01-04T00:30:00.000Z,
          },
        ],
        "IsCODCapable": true,
        "IsFulfillable": true,
        "ShippingSpeedCategory": "Standard",
        "UnfulfillablePreviewItems": Array [],
      },
      Object {
        "EstimatedFees": Array [
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "225",
            },
            "Name": "FBAPerUnitFulfillmentFee",
          },
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "475",
            },
            "Name": "FBAPerOrderFulfillmentFee",
          },
          Object {
            "Amount": Object {
              "CurrencyCode": "JPY",
              "Value": "660",
            },
            "Name": "FBATransportationFee",
          },
        ],
        "EstimatedShippingWeight": Object {
          "Unit": "kilograms",
          "Value": "12",
        },
        "FulfillmentPreviewShipments": Array [
          Object {
            "EarliestArrivalDate": 2014-01-05T07:00:00.000Z,
            "EarliestShipDate": 2014-01-04T00:30:00.000Z,
            "FulfillmentPreviewItems": Array [
              Object {
                "EstimatedShippingWeight": Object {
                  "Unit": "kilograms",
                  "Value": "10",
                },
                "Quantity": 2,
                "SellerFulfillmentOrderItemId": "mws-test-query-20100713023406723-2",
                "SellerSKU": "SampleSKU1",
                "ShippingWeightCalculationMethod": "Package",
              },
              Object {
                "EstimatedShippingWeight": Object {
                  "Unit": "kilograms",
                  "Value": "2",
                },
                "Quantity": 1,
                "SellerFulfillmentOrderItemId": "mws-test-query-20100713023406723-1",
                "SellerSKU": "SampleSKU2",
                "ShippingWeightCalculationMethod": "Package",
              },
            ],
            "LatestArrivalDate": 2014-01-06T06:59:59.000Z,
            "LatestShipDate": 2014-01-04T00:30:00.000Z,
          },
        ],
        "IsCODCapable": false,
        "IsFulfillable": true,
        "ScheduledDeliveryInfo": Object {
          "DeliveryTimeZone": "Asia/Tokyo",
          "DeliveryWindows": Array [
            Object {
              "EndDateTime": 2014-01-05T03:00:00.000Z,
              "StartDateTime": 2014-01-04T23:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-05T05:00:00.000Z,
              "StartDateTime": 2014-01-05T03:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-05T07:00:00.000Z,
              "StartDateTime": 2014-01-05T05:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-05T09:00:00.000Z,
              "StartDateTime": 2014-01-05T07:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-05T11:00:00.000Z,
              "StartDateTime": 2014-01-05T09:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-05T12:00:00.000Z,
              "StartDateTime": 2014-01-05T11:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-06T03:00:00.000Z,
              "StartDateTime": 2014-01-05T23:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-06T05:00:00.000Z,
              "StartDateTime": 2014-01-06T03:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-06T07:00:00.000Z,
              "StartDateTime": 2014-01-06T05:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-06T09:00:00.000Z,
              "StartDateTime": 2014-01-06T07:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-06T11:00:00.000Z,
              "StartDateTime": 2014-01-06T09:00:00.000Z,
            },
            Object {
              "EndDateTime": 2014-01-06T12:00:00.000Z,
              "StartDateTime": 2014-01-06T11:00:00.000Z,
            },
          ],
        },
        "ShippingSpeedCategory": "ScheduledDelivery",
        "UnfulfillablePreviewItems": Array [],
      },
    ],
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": 2020-04-06T10:22:23.582Z,
    "requestId": "0",
    "timestamp": 2020-05-06T09:22:23.582Z,
  },
]
`;

exports[`fulfillmentOutboundShipment getServiceStatus returns a parsed model when the status response is valid 1`] = `
Array [
  Object {
    "Status": "GREEN",
    "Timestamp": "2020-05-06T08:22:23.582Z",
  },
  Object {
    "quotaMax": 1000,
    "quotaRemaining": 999,
    "quotaResetOn": 2020-04-06T10:22:23.582Z,
    "requestId": "0",
    "timestamp": 2020-05-06T09:22:23.582Z,
  },
]
`;
